src/Microsoft.OpenApi.OData.Reader/Operation/DollarCountGetOperationHandler.cs

```diff
@@ -36,6 +39,16 @@ protected override void Initialize(ODataContext context, ODataPath path)
             int count = path.Segments.Count;
             if(count >= SecondLastSegmentIndex)
                 LastSecondSegment = path.Segments.ElementAt(count - SecondLastSegmentIndex);
+
+            if (LastSecondSegment is ODataNavigationSourceSegment sourceSegment)
+            {
+                annotatable = sourceSegment.NavigationSource as IEdmEntitySet;
+                annotatable ??= sourceSegment.NavigationSource as IEdmSingleton;
```

Comment:
```suggestion
                annotatable = sourceSegment.NavigationSource as IEdmEntitySet ?? sourceSegment.NavigationSource as IEdmSingleton;
```

Reply:
Getting an error here: _Operator '??' cannot be applied to operands of type IEdmEntitySet and IEdmSingleton_. Reverting to original code.

Reply:
could we use a common parent interface instead? like IAnnotable or something similar?

Reply:
I have opted to do away with the `IEdmSingleton` cast. We are not using it anyway